core/src/main/scala/org/dbpedia/spotlight/db/similarity/VectorContextSimilarity.scala

```diff
@@ -0,0 +1,72 @@
+package org.dbpedia.spotlight.db.similarity
+
+import org.dbpedia.spotlight.db.memory.MemoryVectorStore
+import org.dbpedia.spotlight.db.model.ContextStore
+
+import org.dbpedia.spotlight.model.{DBpediaResource, TokenType}
+import org.dbpedia.spotlight.util.MathUtil.{cosineSimilarity, LOGZERO}
+
+import scala.collection.mutable
+import scala.util.Random
+
+
+/**
+ * Context similarity based on dense, continuous space vector models.
+ * @author Philipp Dowling
+ *
+ * Created by dowling on 12/06/15.
+ */
+case class VectorContextSimilarity(memoryVectorStore: MemoryVectorStore) extends ContextSimilarity{
+
+  def vectorSimilarities(query: Seq[TokenType], candidates: Set[DBpediaResource]): mutable.Map[DBpediaResource, Double] = {
+    val contextScores = mutable.HashMap[DBpediaResource, Double]()
+
+    candidates.map( resource => {
+      contextScores.put(
+        resource,
+        // similarity of context and current resource
+        // TODO: use the whole context, or only surrounding n words?
```

Comment:
I was wondering if there is a nead way to capture of your TODO in terms of adding an extra argument (a function) that modifies the sequence of tokentypes. By default that function would be Identity, but we could pass different ones to measure the impact on performance